File: scripts/print_disasm.py
# GhidraScript (Jython): print all instructions and functions found.
# Run via: analyzeHeadless ... -postScript print_disasm.py
from ghidra.app.util.headless import HeadlessScript
from ghidra.program.model.listing import CodeUnit

listing = currentProgram.getListing()
func_mgr = currentProgram.getFunctionManager()
lang = currentProgram.getLanguage()

print("=" * 60)
print("Language:", lang.getLanguageID())
print("=" * 60)

# List all functions
functions = list(func_mgr.getFunctions(True))
print(f"\nFunctions found: {len(functions)}")
for fn in functions:
    entry = fn.getEntryPoint()
    print(f"  {fn.getName():<20} @ {entry}")

# Dump all instructions
print("\nDisassembly:")
instr_iter = listing.getInstructions(True)
count = 0
while instr_iter.hasNext() and count < 100:
    instr = instr_iter.next()
    addr = instr.getAddress()
    mnemonic = instr.getMnemonicString()
    ops = instr.getDefaultOperandRepresentationList(0) if instr.getNumOperands() > 0 else []
    print(f"  {addr}  {mnemonic:<10} {instr.toString().split(mnemonic, 1)[-1].strip()}")
    count += 1

print(f"\nTotal instructions disassembled: {count}")
print("=" * 60)

File: scripts/PrintDisasm.java
// @category Analysis
import ghidra.app.script.GhidraScript;
import ghidra.program.model.listing.*;
import ghidra.program.model.address.AddressSetView;

public class PrintDisasm extends GhidraScript {
    @Override
    public void run() throws Exception {
        println("========================================");
        println("Language: " + currentProgram.getLanguageID());
        println("Entry:    " + currentProgram.getImageBase());

        FunctionManager fm = currentProgram.getFunctionManager();
        FunctionIterator funcs = fm.getFunctions(true);
        int nFuncs = 0;
        println("\nFunctions:");
        while (funcs.hasNext()) {
            Function fn = funcs.next();
            println("  " + fn.getEntryPoint() + "  " + fn.getName());
            nFuncs++;
        }
        println("Total functions: " + nFuncs);

        println("\nDisassembly:");
        Listing listing = currentProgram.getListing();
        InstructionIterator ii = listing.getInstructions(true);
        int count = 0;
        while (ii.hasNext() && count < 50) {
            Instruction instr = ii.next();
            println("  " + instr.getAddressString(false, true) + "  " + instr.toString());
            count++;
        }
        println("Total instructions: " + count);
        println("========================================");
    }
}
